File: dsa_project_gaurav/Ques4.java
package dsa_project_gaurav;

public class Ques4 {
    public static void main(String[] args) {
        int[] nums = { 0, 1, 1, 3, 3, 5, 5, 5, 9, 9, 9, 9, 10, 10, 10, 10, 10 };

        int k = removeDuplicates(nums);

        System.out.print("Unique elements: ");
        for (int i = 0; i < k; i++) {
            System.out.print(nums[i] + " ");
        }
        System.out.println("\nCount of unique elements = " + k);
    }

    public static int removeDuplicates(int[] nums) {
        if (nums.length == 0)
            return 0;

        int i = 0;

        for (int j = 1; j < nums.length; j++) {
            if (nums[j] != nums[i]) {
                i++;
                nums[i] = nums[j];
            }
        }

        return i + 1;
    }
}

/*
This code has a function in which the first number will points the last unique element and the loop in it will find whether the number taken has any similar number in the array or not 
if it contain then it will be renamed coz in the nothing can be deleted but can be reassigned it's value and last the function has been return.
Now, this function will be called in the main function which also contains a k times of unique element.
At the last the print function which will print my actual removed duplicate from sorted array will be printed.

HAPPY CODING !!!...
*/

File: DSA project (Hamid)/Ques4.py
def remove_duplicates(nums):
    if not nums:
        return 0
    
    k = 1  # This is pointer for the next unique element
    
    for i in range(1, len(nums)):
        if nums[i] != nums[i - 1]:
            nums[k] = nums[i]
            k += 1
    
    return k

# Example usage
nums = [1, 1, 2]
k = remove_duplicates(nums)
print(k)        # Output: 2
print(nums[:k]) # Output: [1, 2]
